@patch.object(jenkins.Jenkins, 'jenkins_open')
def test_failed(self, jenkins_mock):
    jenkins_mock.side_effect = [json.dumps({'name': 'TestView', }), json.dumps({'name': 'TestView', }), json.dumps({'name': 'TestView', })]
    with self.assertRaises(jenkins.JenkinsException) as context_manager:
        self.j.delete_view(u'TestView')
    self.assertEqual(jenkins_mock.call_args_list[0][0][0].get_full_url(), self.make_url('view/TestView/doDelete'))
    self.assertEqual(str(context_manager.exception), 'delete[TestView] failed')
    self._check_requests(jenkins_mock.call_args_list)
